### ISSUE TITLE
Icons: standardize images

### ISSUE THREAD

**Issue by justintemps** [international-labour-organization/designsystem#839]
Hi @inesdgomes, we finally fixed the icon rendering in the button and other places. However, while I was testing it, it seems like some of the icons have much different dimensions and thus render differently inside components that use them, like buttons.

Here are a few examples:

<img width="146" alt="image" src="https://github.com/international-labour-organization/designsystem/assets/12401179/1b41f6b4-4309-411b-83d3-63ee9e15a081">

<img width="143" alt="image" src="https://github.com/international-labour-organization/designsystem/assets/12401179/d62978db-3903-49de-9519-d503f4ac5929">

<img width="146" alt="image" src="https://github.com/international-labour-organization/designsystem/assets/12401179/0d1dda60-ee5d-4267-8cd6-b4f1e6c4d9ca">

<img width="141" alt="image" src="https://github.com/international-labour-organization/designsystem/assets/12401179/bb1ae26d-9805-4f94-aa0d-6548b8624503">

I added a table of all of the icons with the names and previews to that pacakge README which you can [see here](https://github.com/international-labour-organization/designsystem/tree/develop/packages/icons)

I think the problem is that many of the icons have differing amounts of padding in their viewBoxes. The SVGs themselves are the same size, but there are varying amounts of empty space around the actual image inside the SVGs that cause the icons to appear as if they're different sizes.

I think the solution here is to do the same thing we did with the logos, recreating them one by one to make sure that they all have the same proportion.

**Comment by inesdgomes** [comment]
Hi @justintemps OK, will change the icons. Before I do that, does this make sense to you:
- All SVGs are 24:24px (no exceptions)
- Icons inside the SVGs are always centered
- Icons inside the SVGs are always either 16px wide or 16px tall, depending on the icon (this is what we did for the social media icons)
- Icons with very odd ratios may exceptionally be a bit taller or wider - this may be necessary for icons that are very long or tall, but will try to avoid the exceptions.

**Comment by justintemps** [comment]
- Hey @inesdgomes, I'm not sure why we'd want the SVG to be 24px but the image inside the SVG to only be 16px. They'll be much easier to work with if they go edge to edge.
- Yes, they should always be centered.

**Comment by justintemps** [comment]
We agreed that @inesdgomes would standardize the icons in Figma and then we'll figure out a way to export them to the package

**Comment by beatrizmartinmartins** [comment]
@inesdgomes let's talk about a few of these icons, because I think it doesn't make sense to have a few of them and we can deprecate them and use similar ones. Like for example, why do we have so many types of chevrons? we should use only one to make sure the experience looks cohesive. Right now, from my point of view, its like we are using two different sets of icons. 

![Image](https://github.com/international-labour-organization/designsystem/assets/96917981/580d861d-0808-44f5-8ed6-5cafa0a3bd26)

![Image](https://github.com/international-labour-organization/designsystem/assets/96917981/685de201-6fe1-49e6-81a3-55b39ecd8174)

![Image](https://github.com/international-labour-organization/designsystem/assets/96917981/9a3feef3-8522-49dd-a9df-be097a1929c3)

Also, a few of the icons have border-radius on some corners and others don't. Shouldn't we use the same style for all of them?

**Comment by inesdgomes** [comment]
Hi @beatrizmartinmartins As discussed, please do streamline the style of the icons and remove duplicates and icons that are not used anywhere. Thanks

**Comment by beatrizmartinmartins** [comment]
Please check the updates [here](https://www.figma.com/design/szl87D4Zg4MeWXzKWMCCZO/ILO-Icons?node-id=5949-9509&t=Flt7UX1axh8CRE2O-1) All of the icons have been updated to the latest style and we don't have duplicates anymore

**Comment by inesdgomes** [comment]
@beatrizmartinmartins Much better, thanks! I remember you pointed out that some icons also had different styles, for example the home icon has rounded corners. Please could you also review that and then list here which icons were changed so developers can replace them? Thanks

**Comment by beatrizmartinmartins** [comment]
@inesdgomes All those icons are review and I just changed them. Honestly, all icons have changed. I changed every single one of them. Adjusted the sizings, the paddings, made sure that every single icon looked nice and applied it to templates to make more reviews. The list of final icons is the following:

- Add
- Close
- Minus
- Caret Right
- Caret Left
- Caret Down
- Caret Up
- Arrow Left
- Arrow Right
- Arrow Down
- Arrow Up
- Arrow Go To Right
- Arrow Go To Left
- Home
- Arrow Sort
- Chevron Down
- Chevron Up
- Chevron Right
- Chevron Left
- Download
- Calendar
- Search
- Check
- Copyright
- Send
- Mail
- Launch
- Online
- Conference
- Dot Menu
- Notification Success
- Notification Error
- Notification Info
- Notification Info Outlined
- Notification Warning
- Burger
- Language
- Video Sound On
- Video Sound Off
- Settings
- Fullscreen
- Pause
- Play
- Statistics
- Social RSS
- Pin
- Social TikTok
- Social LinkedIn
- Social Facebook
- Social Twitter
- Social Instagram
- Social Flickr
- Social Youtube
- Upload

cc @justintemps

**Comment by inesdgomes** [comment]
Ok, cool! One final question: is there a reason for the right and left chevrons to be smaller than the other chevrons? Wondering how that's going to look in the search results' pagination? Scroll to the bottom here: https://www.ilo.org/search?search_api_fulltext=&ilo_date%5Bmin%5D=&ilo_date%5Bmax%5D=&sort_by=search_api_relevance&f%5B0%5D=language%3Aen 
<img width="866" alt="Screenshot 2024-06-10 at 13 07 27" src="https://github.com/international-labour-organization/designsystem/assets/22772102/95ec1cd0-1f41-4a6b-9624-99b71fd9dd17">

**Comment by beatrizmartinmartins** [comment]
@inesdgomes so basically left and right chevrons are used to navigate through content or pagesm basically an action, while the top and bottom chevrons is more of a dropdown kind of chevron where it shows something up, not really a navigation. Although if you'd like we can make them the same size and have them all look the same

**Comment by beatrizmartinmartins** [comment]
Review and sizing aligned @inesdgomes

**Comment by inesdgomes** [comment]
Thanks @beatrizmartinmartins Could you just check the chevrons in the templates? Something doesn't look quite right in Search Results pages.

![Image](https://github.com/international-labour-organization/designsystem/assets/22772102/7cfd5875-fb15-40bb-8db4-68c6bcfe349f)

**Comment by beatrizmartinmartins** [comment]
@inesdgomes checked and changed on the templates!

**Comment by inesdgomes** [comment]
Hi @justintemps This is ready for development: https://www.figma.com/design/szl87D4Zg4MeWXzKWMCCZO/ILO-Icons?node-id=5949-9510&t=ttLvr2f80hUvsoWB-0 

Beatriz adjusted all the icons and removed duplicates and unnecessary icons. All icons need to be replaced in the code.

**Comment by inesdgomes** [comment]
@justintemps This is ready for development. See comment above.

**Comment by justintemps** [comment]
@Shashika6 so the task here is to replace the icon files that we have in the icon packages with the ones we have above and include any that are missing (including download).

**Comment by inesdgomes** [comment]
@Shashika6 Please replace all SVG files, even those that look similar. All icons were changed, even if by a few pixels.

**Comment by Shashika6** [comment]
Hey team can I please get access to this one. https://www.figma.com/design/szl87D4Zg4MeWXzKWMCCZO/ILO-Icons?node-id=5949-9510&t=ttLvr2f80hUvsoWB-0

![Image](https://github.com/user-attachments/assets/af4666a8-1ffb-47c7-9b46-a7b3d0d5dc6f)


email is :- Shashika.boteju@nuvole.org

Cc :- @inesdgomes @justintemps

**Comment by inesdgomes** [comment]
Hi @Shashika6 Sorry, thought you had access to all the files. It should work now.

**Comment by Shashika6** [comment]
Hey team so I started working on this one and I noticed a few things and have some questions. 

* Some of the currently existing icons that are being used have different names should we change these to sync with the new names on the design ?

* Some of the old icons are not available in the designs but they are used on the website. How are we going to action this ?

* There are a few new icons should we add them into the package as well ?

**Comment by justintemps** [comment]
Discussed this with @inesdgomes and we agreed that we should change the names of the icons to the ones that are used in Figma. Also, we need a list of the icon names to the Storybook documentation for the Icon component, otherwise you don't understand how to use it.

**Comment by inesdgomes** [comment]
Hi @Shashika6 Just to clarify:

- _Some of the currently existing icons that are being used have different names should we change these to sync with the new names on the design ?_ **Yes, we should use the names from the design files**

_- _- Some of the old icons are not available in the designs but they are used on the website. How are we going to action this_ Can you please let me know which icons are these?

- There are a few new icons should we add them into the package as well ?_ In principle, yes, but could you let me know which ones are new?

**Comment by Shashika6** [comment]
Sure @inesdgomes will document everything on the go.

**Comment by Shashika6** [comment]
Hey @justintemps @inesdgomes. 

I documented how the new icons are going to be. 

As mentioned, it's clear how to add and replace the new icons and the unused ones can be easily removed, but what about the existing ones that are not in the design? 

### New Icons added

ArrowGoToLeft
ArrowSort
Calendar
Upload
NotificationInfo Conference
Settings
Statistics Launch
Online Mail
Pin
Send
Share
ClosedCaptions ChevronUp
ChevronLeft
ChevronRight
CaretUp
CaretLeft


### Replaced Icons

Add
Arrow -> ArrowDown
ArrowUp
BreadcrumbDividerRTL -> ArrowLeft
PaginationArrow/ BreadcrumbDivider -> ArrowRight 
Burger
CheckMark -> Check
Download
DoubleArrow-> ArrowGoToRight	
Home
Copyright
Play Pause
Facebook -> SocialFacebook
Instagram -> SocialInstagram
Flickr -> SocialFlickr
RSS -> SocialRss
X -> SocialX
LinkedIn -> SocialLinkedIn
TikTok -> SocialTiktok
Elipsis -> DotMenu
Global -> Language Error -> NotifcationError Info -> NotificationInfoOutlined
Success -> NotificationSuccess
Warning -> NotificationWarning
Search 
Minus
Muted -> VideoSoundOff Volume -> VideoSoundOn
SteMarrow -> ChevronDown
ListArrow-> CaretRight
Equilateraltriangle -> CaretDown
 
### Used Old icons not available on the design 

Quote
FooterTriangle
HalfSquareTriangle
Triangle

### Unused Old icons 

Scalenerighttriangle
Scalenerighttrianglereverse
TriangleDecoration
TriangleReverse
ListArrowReverse

**Comment by justintemps** [comment]
Hey @Shashika6 

We need to make sure that the Unused Old Icons aren't being used by the website somewhere.

@inesdgomes 

As per @Shashika6's comment above, we have some old icons that are not in the designs but that are being used in various places by the DS. @Shashika6 will paste some examples below.

## Used Old icons not available on the design

Quote
FooterTriangle
HalfSquareTriangle

**Comment by inesdgomes** [comment]
Hi @Shashika6 Thanks for this. @justintemps On the old icons that are used in the DS but are not available in the design, I'll discuss with @beatrizmartinmartins, but my first thought is:
- Quote should be added to the Icons package in the designs file.
- FooterTriangle and HalfSquareTriangle do not have the value of icons and they could perhaps be moved to the [brand assets package](https://github.com/international-labour-organization/designsystem/tree/develop/packages/brand-assets). I don't think we need them in the designs files, but will check with Beatriz.

As for the Unused Old icons, if we're 100% sure they're not used anywhere, we can delete them.

**Comment by beatrizmartinmartins** [comment]
Are [these](https://www.figma.com/design/szl87D4Zg4MeWXzKWMCCZO/ILO-Icons?node-id=1896-30&t=4POWBnyhDnPjuzo3-1) ones the ones missing from this file? @Shashika6 Is it possible to use change the names of them to something more agnostic?
- FooterTriangle could be Horizontal Triangle
- HalfSquareTriangle could be Triangle
Something like this maybe?

Should I add these Icons to the components where they are being used? @inesdgomes

**Comment by Shashika6** [comment]
Yes @beatrizmartinmartins :) those are the icons that are missing. They seem to be different in size compared to the other ones as well. 

Also, Yes we can change the names of these icons.

**Comment by Shashika6** [comment]
Hey team waiting on a reply to this :) 

Cc :- @beatrizmartinmartins , @inesdgomes

**Comment by inesdgomes** [comment]
Hi @Shashika6 Beatriz and I talked about this and agreed that:
- We'll keep only the 'Quote' and 'HalfSquareTriangle' icons. Beatriz already added them to the design files in a separate page called [Others](https://www.figma.com/design/szl87D4Zg4MeWXzKWMCCZO/ILO-Icons?node-id=1896-30&node-type=canvas&t=hsAf32p7rzAVlVg2-0).
- As we're getting rid of the 'FooterTriangle', we'll use the 'HalfSquareTriangle' in the footer instead. Beatriz will update the component.
- Beatriz will also make sure that in the design files we use the 'HalfSquareTriangle' in the credits and context menu, to ensure it's aligned with development.
- As for the name, @beatrizmartinmartins and @Shashika6 let's use 'Triangle-RightAngle'. I know it's long, but we need to differentiate it from the equilateral triangle, which is part of the ILO brand guidelines.

**Comment by Shashika6** [comment]
Sure @inesdgomes, Thanks :'). @beatrizmartinmartins Let me know when you update the above component.

**Comment by beatrizmartinmartins** [comment]
Hey, we are using the long triangle on the Cards as well. Should we use the Triangle-RightAngle instead @inesdgomes ?

**Screenshot attached**
![Image](https://github.com/user-attachments/assets/cf8c9a36-2bd1-4419-b5c4-28133a489faa)

**Comment by beatrizmartinmartins** [comment]
Apart from the above, all the other components like Footer, Credits and Context Menu, as well as the Strips of the NavBar have been updated. @Shashika6 and @inesdgomes Check them here:

- [Credit](https://www.figma.com/design/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?m=auto&node-id=6298-11285&t=JIVvLQYx1iQ47Ijn-1)
- [Footer](https://www.figma.com/design/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?m=auto&node-id=1590-15140&t=JIVvLQYx1iQ47Ijn-1)
- [Navigation Strips](https://www.figma.com/design/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?m=auto&node-id=6198-11143&t=JIVvLQYx1iQ47Ijn-1)

**Comment by inesdgomes** [comment]
> Hey, we are using the long triangle on the Cards as well. Should we use the Triangle-RightAngle instead @inesdgomes ?
> 
> **Screenshot attached** ![Image](https://private-user-images.githubusercontent.com/96917981/372153585-cf8c9a36-2bd1-4419-b5c4-28133a489faa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3NzI3OTMsIm5iZiI6MTcyNzc3MjQ5MywicGF0aCI6Ii85NjkxNzk4MS8zNzIxNTM1ODUtY2Y4YzlhMzYtMmJkMS00NDE5LWI1YzQtMjgxMzNhNDg5ZmFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAxVDA4NDgxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyZTdlMDJhOWI0NmY0N2Q4MTViOGZhMWM5OTE0NDhiNjE4Y2Q2Mjg3YjIxZDViYjFlMzdkMjY0YTNiMmNkMTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8yq3TGAI3RYaE6B2dzHygOLa8IHl_uyV9umVD2KqBWI)

Hi @beatrizmartinmartins Can we see how it would look with the Triangle-RightAngle?

**Comment by inesdgomes** [comment]
> Apart from the above, all the other components like Footer, Credits and Context Menu, as well as the Strips of the NavBar have been updated. @Shashika6 and @inesdgomes Check them here:
> 
> * [Credit](https://www.figma.com/design/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?m=auto&node-id=6298-11285&t=JIVvLQYx1iQ47Ijn-1)
> * [Footer](https://www.figma.com/design/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?m=auto&node-id=1590-15140&t=JIVvLQYx1iQ47Ijn-1)
> * [Navigation Strips](https://www.figma.com/design/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?m=auto&node-id=6198-11143&t=JIVvLQYx1iQ47Ijn-1)

Thanks @beatrizmartinmartins Some comments:
- [Credit](https://www.figma.com/design/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?m=auto&node-id=6298-11285&t=JIVvLQYx1iQ47Ijn-1) There's a small size issue here. I left a comment in the design
[Footer](https://www.figma.com/design/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?m=auto&n
<img width="1658" alt="Screenshot 2024-10-01 at 10 56 08" src="https://github.com/user-attachments/assets/72db914c-15a8-4e3b-90c6-fc339179b47b">
ode-id=1590-15140&t=JIVvLQYx1iQ47Ijn-1) The triangle in the footer creates weird layouts, depending on the breakpoint and amount of social media icons. This was happening with the other triangle too. See screenshot below of the footer in the ILO global website. Should we make it much smaller so it uses just a tiny bit of the corner? Or would you rather completely remove it?
[Navigation Strips](https://www.figma.com/design/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?m=auto&node-id=6198-11143&t=JIVvLQYx1iQ47Ijn-1) All good

- [Context menu ](https://www.figma.com/design/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?node-id=6008-9462&node-type=frame&t=zMXIi3Xw1lbyceFN-0) This component is using another triangle, which is an equilateral one. The equilateral one should also be used in the lists, though it has been replaced with the carret. Let's talk about this tomorrow.

**Comment by beatrizmartinmartins** [comment]
@inesdgomes I think it doesn't really make much difference using one triangle or another honestly, so I'd go with the Triangle-RightAngle that we decided to keep. What do you think?
**Before**
![Image](https://github.com/user-attachments/assets/1c0d3d4f-9cfb-47ac-8698-0f0fbfa882b1)

**After**
![Image](https://github.com/user-attachments/assets/2e28c36c-a6bd-463b-8d6d-cb51fe4189d4)


Related to the components:

- Credit. Already fixed it.
- Footer. I think we should remove it. It doesn't add anything to the footer anyway and if it's making our lives harder, it's better to remove.
- Context Menu. I think that is not even an icon, is simply a triangle container. But yes, let's talk about it tomorrow.

**Comment by inesdgomes** [comment]
@beatrizmartinmartins Agree on changing the triangle in the card component, as well as removing the triangle from the footer.
To discuss tomorrow:
- Should we also remove the triangle from the credit button? It does look odd and it's not making things easy in the photo gallery component.
- Triangle in the context menu and lists. Do we need an equilateral triangle for those?

**Comment by justintemps** [comment]
hey @Shashika6 I don't think the above request is necessarily ready for dev yet, please hold off on that.

**Comment by justintemps** [comment]
Hey @beatrizmartinmartins and @inesdgomes, I think that in the Figma files you guys mixed up arrows and chevrons.

This is an arrow:

![arrowleft](https://github.com/user-attachments/assets/7bbd4c93-13cb-4d6e-8141-bc40e8f4f40b)

And this is a chevron:

![chevronleft](https://github.com/user-attachments/assets/f3608493-18c2-41c3-95e4-dda28ab7cb40)

I've renamed these appropriately in the in the icons package. Here's a brief recap of all of the icon names:

```
add.svg
arrowdown.svg
arrowleft.svg
arrowright.svg
arrowsort.svg
arrowup.svg
burger.svg
calendar.svg
caretdown.svg
caretleft.svg
caretright.svg
caretup.svg
check.svg
chevrondown.svg 
chevrongoleft.svg  <- Used to be arrowgotoleft.svg
chevrongoright.svg <- Used to be arrowgotoright.svg
chevronleft.svg
chevronright.svg
chevronup.svg
close.svg
closedcaptions.svg
conference.svg
copyright.svg
dotmenu.svg
download.svg
footertriangle.svg
fullscreen.svg
halfsquaretriangle.svg
home.svg
language.svg
launch.svg
mail.svg
minus.svg
notificationerror.svg
notificationinfo.svg
notificationinfooutlined.svg
notificationsuccess.svg
notificationwarning.svg
online.svg
pause.svg
pin.svg
play.svg
quote.svg
search.svg
send.svg
settings.svg
share.svg
socialfacebook.svg
socialflickr.svg
socialinstagram.svg
sociallinkedin.svg
socialrss.svg
socialtiktok.svg
socialx.svg
socialyoutube.svg
statistics.svg
upload.svg
videosoundoff.svg
```

**Comment by inesdgomes** [comment]
Hi @beatrizmartinmartins A few notes on this for our meeting:

- I already corrected the names of arrows/chevrons
- We need to check which triangle is being used in lists and context menu. Ideally, it should be an equilateral triangle.
- The instagram icon is the only social media icon that is outlined. Should we change that?

**Comment by beatrizmartinmartins** [comment]
Hey @inesdgomes and @justintemps 

I just included the icons listed above:

- Changed all the carets to have an equilateral triangle (based on what we have on the Brand Guidelines defined)
- Removed the Play button triangle. We are going to use the Caret from now on.
- Added the Triangle-Equilateral to be used on tooltips, loading and context menus.
- X (Previously Twitter) icon doesn't have an option with solid color, so we will keep it like it is.
- Instagram logo has been changed from outlined to solid form.
